starter_code/queries.md

```diff
@@ -1,83 +1,104 @@
 
-<!-- Your Code Goes Here -->
 
-### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
+### 9. Order all the companies by their IPO price descendently.
 
-<!-- Your Code Goes Here -->
+F:{ipo:{$ne:null}}
+S:{ipo:-1}
 
-### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-<!-- Your Code Goes Here -->
+### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
-<!-- ### 12. All the companies that have been 'deadpooled' after the third year. -->
+F:{number_of_employees: {$gte:0}}
+S:{number_of_employees:-1}
+L:10
 
-<!-- Your Code Goes Here -->
+### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
-### 12. All the companies founded before 2000 that have and acquisition amount of more than 10.000.000
+<!-- Your Code Goes Here -->
 
-<!-- Your Code Goes Here -->
+<!-- ### 12. All the companies that have been 'deadpooled' after the third year. -->
 
-### 13. All the companies that have been acquired after 2010, order by the acquisition amount, and retrieve only their `name` and `acquisition` field.
+<!-- Your Code Goes Here -->
 
-<!-- Your Code Goes Here -->
+### 12. All the companies founded before 2000 that have and acquisition amount of more than 10.000.000
 
-### 14. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
+F: {founded_year:{$lt:2000}, "acquisition.price_amount":{$gte:10000000}}
 
-<!-- Your Code Goes Here -->
+### 13. All the companies that have been acquired after 2010, order by the acquisition amount, and retrieve only their `name` and `acquisition` field.
 
-### 15. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `acquisition price` descendently. Limit the search to 10 documents.
+F:{"acquisition.acquired_year":{$gte:2010}}
+P:{"acquisition.price_amount":1,name:1,_id:0}
+S:{"acquisition.price_amount":1}
 
-<!-- Your Code Goes Here -->
+### 14. Order the companies by their `founded year`, retrieving only their `name` and `founded year`.
 
-### 16. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascendant order.
+F:{founded_year:1,name:1,_id:0}
+S:{founded_year:1}
 
-<!-- Your Code Goes Here -->
+### 15. All the companies that have been founded on the first seven days of the month, including the seventh. Sort them by their `acquisition price` descendently. Limit the search to 10 documents.
 
-### 17. All the companies which their acquisition amount is more than 10.000.000, and currency are 'EUR'.
+F:{founded_day: {$lte: 7}}
+S: {"acquisitions.price_amount": -1}
 
-<!-- Your Code Goes Here -->
+### 16. All the companies on the 'web' `category` that have more than 4000 employees. Sort them by the amount of employees in ascendant order.
```

Comment:
query: {$and:[{category_code:"web"},{number_of_employees:{$gt:4000}}]}
sort: {number_of_employees:-1}